File: mobile/src/main/java/com/windscribe/mobile/newsfeedactivity/NewsFeedView.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.newsfeedactivity;

import com.windscribe.mobile.adapter.NewsFeedAdapter;
import com.windscribe.vpn.api.response.PushNotificationAction;


public interface NewsFeedView {

    void hideProgress();

    void setNewsFeedAdapter(NewsFeedAdapter mAdapter);

    void showLoadingError(String errorMessage);

    void showProgress(String progressTitle);

    void startUpgradeActivity(PushNotificationAction pushNotificationAction);
}

File: mobile/src/main/java/com/windscribe/mobile/newsfeedactivity/NewsFeedView.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.newsfeedactivity

import com.windscribe.mobile.adapter.NewsFeedAdapter
import com.windscribe.vpn.api.response.PushNotificationAction

interface NewsFeedView {
    fun hideProgress()
    fun setNewsFeedAdapter(mAdapter: NewsFeedAdapter)
    fun showLoadingError(errorMessage: String)
    fun showProgress(progressTitle: String)
    fun startUpgradeActivity(pushNotificationAction: PushNotificationAction)
}